From 5e6c614d8e7cf406a38dcdfd4b4e73c1cbb6a4a8 Mon Sep 17 00:00:00 2001
From: Varun Berry <vberry@google.com>
Date: Fri, 5 Nov 2021 11:04:39 -0700
Subject: [PATCH 1/5] [Q_asb_2022-01] Add FLAG_IMMUTABLE to PendingIntent in
 LegacyModeSmsHandler.

Bug: 185126549, 185126319
Test: manual.

Merged-In: I1024353434674f8e4bf164c3a6c1e16924dfaa8f
Change-Id: I3b015de3187f45096235313e8d39a3c3f1b768b7
(cherry picked from commit bdd6d1ea00f94296c251f4340a52a2035167370b)
Merged-In: I3b015de3187f45096235313e8d39a3c3f1b768b7
---
 java/com/android/voicemail/impl/sms/LegacyModeSmsHandler.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/java/com/android/voicemail/impl/sms/LegacyModeSmsHandler.java b/java/com/android/voicemail/impl/sms/LegacyModeSmsHandler.java
index 29d376b35..4e8272207 100644
--- a/java/com/android/voicemail/impl/sms/LegacyModeSmsHandler.java
+++ b/java/com/android/voicemail/impl/sms/LegacyModeSmsHandler.java
@@ -109,7 +109,7 @@ public class LegacyModeSmsHandler {
                   context,
                   CallIntentBuilder.forVoicemail(
                       phoneAccountHandle, CallInitiationType.Type.LEGACY_VOICEMAIL_NOTIFICATION)),
-              PendingIntent.FLAG_UPDATE_CURRENT);
+              PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
     } else {
       Intent launchVoicemailSettingsIntent =
           new Intent(TelephonyManager.ACTION_CONFIGURE_VOICEMAIL);
@@ -122,7 +122,7 @@ public class LegacyModeSmsHandler {
               context,
               LAUNCH_VOICEMAIL_SETTINGS_REQUEST_CODE,
               launchVoicemailSettingsIntent,
-              PendingIntent.FLAG_UPDATE_CURRENT);
+              PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
     }
 
     intent.putExtra(TelephonyManager.EXTRA_VOICEMAIL_NUMBER, voicemailNumber);
-- 
2.35.1

